scripts/rmc_flags.py

```diff
@@ -0,0 +1,136 @@
+             help="""
+                  The working directory. Used to determine source locations in output.
+                  This is generally the location from which rmc is currently being invoked.
+                  """)
+    add_flag(group, "--visualize", action="store_true",
+             help="Generate visualizer report; open report/html/index.html")
+
+# Add flags for ad-hoc features.
+def add_other_flags(make_group, add_flag):
+    group = make_group("Other flags")
+    add_flag(group, "--allow-cbmc-verification-failure", action="store_true",
+             help="Do not produce error return code on CBMC verification failure")
+    add_flag(group, "--mangler", default="v0",
+             help="Change what mangler is used by the Rust compiler")
+    add_flag(group, "--dry-run", action="store_true",
+             help="Print commands instead of running them")
+
+# Add flags we don't expect end-users to use.
+def add_developer_flags(make_group, add_flag):
+    group = make_group(
+        "Developer flags", "These are generally meant for use by RMC developers, and are not stable.")
+    add_flag(group, "--cbmc-args", nargs=argparse.REMAINDER, default=[],
+             help="Pass through directly to CBMC; must be the last flag")
+
+# Adds the flags common to both rmc and cargo-rmc.
+# Allows you to specify flags/groups of flags to not add.
+# This does not return the parser, but mutates the one provided.
+def add_flags(parser, exclude_flags=[], exclude_groups=[]):
+    # Keep track of what excluded flags and groups we've seen
+    # so we can warn for possibly incorrect names passed in.
+    excluded_flags = set()
+    excluded_groups = set()
+
+    # Add a group to the parser with title/description, and get a handler for it.
+    def make_group(title=None, description=None):
+        if title in exclude_groups:
+            excluded_groups.add(group.title)
+            return None
+
+        return parser.add_argument_group(title, description)
+
+    # Add the flag to the group, 
+    def add_flag(group, flag, *args, **kwargs):
+        if group == None:
+            return
+
+        if flag in exclude_flags:
+            excluded_flags.add(flag)
+            return
+
+        group.add_argument(flag, *args, **kwargs)
+
+    add_groups = [
+        add_loudness_flags,
+        add_proof_harness_flags,
+        add_artifact_flags,
+        add_check_flags,
+        add_visualizer_flags,
+        add_other_flags,
+        add_developer_flags
+    ]
+
+    for add_group in add_groups:
+        add_group(make_group, add_flag)
+
+    # Error if any excluded flags/groups don't exist.
+    extra_groups = set(exclude_groups) - excluded_groups
+    if extra_groups:
+        print(f"ERROR: Attempt to exclude groups which don't exist: {extra_groups}")
+
+    extra_flags = set(exclude_flags) - excluded_flags
+    if extra_flags:
+        print(f"ERROR: Attempt to exclude flags which don't exist: {extra_flags}")
```

Comment:
Raise an error instead; "Error setting up command line parser..." + link to issue page for users to report bug.